src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/BreakEndVariantType.java

```diff
@@ -40,21 +61,51 @@ public final String toString() {
         this.isTheUpstreamMate = isTheUpstreamMate;
     }
 
-    private static Allele constructAltAllele(final byte[] bases, final boolean bracketPointsLeft, final SimpleInterval novelAdjRefLoc,
-                                             final boolean basesFirst) {
-        final String s = bracketPointsLeft ? "]" + novelAdjRefLoc.getContig() + ":" + novelAdjRefLoc.getStart() + "]"
-                                           : "[" + novelAdjRefLoc.getContig() + ":" + novelAdjRefLoc.getStart() + "[";
-        return Allele.create( basesFirst ? new String(bases) + s  : s + new String(bases) );
-    }
-
-    static abstract class StrandSwitchBND extends BreakEndVariantType {
+    //==================================================================================================================
+    public static final class InvSuspectBND extends BreakEndVariantType {
+        /**
+         * for breakends, there's a concept of partner (see VCF spec) relationship between two reference bases.
+         * This indicates if the breakpoint, for an inversion suspect specifically, represents (one of the two) bases
+         * that is the left of the pair.
+         */
+        private static final Map<String, String> INV55_FLAG = Collections.singletonMap(INV55, "");
+        private static final Map<String, String> INV33_FLAG = Collections.singletonMap(INV33, "");
 
-        StrandSwitchBND(final String id, final Map<String, String> typeSpecificExtraAttributes,
-                        final byte[] bases, final boolean bracketPointsLeft, final SimpleInterval novelAdjRefLoc,
-                        final boolean basesFirst, boolean isTheUpstreamMate){
-            super(id, typeSpecificExtraAttributes, bases, bracketPointsLeft, novelAdjRefLoc, basesFirst, isTheUpstreamMate);
+        private InvSuspectBND(final String id, final byte[] bases, final SimpleInterval novelAdjRefLoc,
+                              final boolean bracketPointsLeft, final boolean basesFirst, boolean isTheUpstreamMate) {
+            super(id, (bracketPointsLeft && basesFirst) ? INV55_FLAG: INV33_FLAG,
+                    bases, bracketPointsLeft, novelAdjRefLoc, basesFirst, isTheUpstreamMate);
         }
 
+        public static Tuple2<BreakEndVariantType, BreakEndVariantType> getOrderedMates(final NovelAdjacencyReferenceLocations narl,
+                                                                                       final ReferenceMultiSource reference) {
+
+            final BreakEndVariantType bkpt_1, bkpt_2;
+            if (narl.strandSwitch == StrandSwitch.FORWARD_TO_REVERSE) { // INV55, leftHalfInPartnerPair
+                bkpt_1 = new BreakEndVariantType.InvSuspectBND(getIDString(narl, true),
```

Comment:
These two clauses are identical except for the three boolean parameters -- it seems like you could just have one set of lines with a boolean to control those.

Reply:
done. Thanks!